### PR TITLE
spec: add missing python3-pyyaml

### PR DESCRIPTION
**Pull request by sebastian-philipp**
https://github.com/ceph/ceph/pull/32817#discussion_r380701967

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dsavineau** [comment]
nothing about `prettytable` ?

**Comment by sebastian-philipp** [comment]
> nothing about `prettytable` ?

`ceph-common` requires prettytable already. Do you think we need this here as well?

**Comment by dsavineau** [comment]
> > nothing about `prettytable` ?
> 
> `ceph-common` requires prettytable already. Do you think we need this here as well?

if it's already requires by ceph-common and mgr-dashboard requires ceph-common why prettytable has been added to the dashboard requirements.txt then ?

**Comment by sebastian-philipp** [comment]
> > > nothing about `prettytable` ?
> > 
> > 
> > `ceph-common` requires prettytable already. Do you think we need this here as well?
> 
> if it's already requires by ceph-common and mgr-dashboard requires ceph-common why prettytable has been added to the dashboard requirements.txt then ?

`pip install -e dashboard/requirements.txt` excludes packages installed by the package manager.

File: ceph.spec.in [hunk 1]
```diff
@@ -493,10 +493,12 @@ Requires:       python%{python3_pkgversion}-six
 %if 0%{?fedora} || 0%{?rhel}
 Requires:       python%{python3_pkgversion}-cherrypy
 Requires:       python%{python3_pkgversion}-werkzeug
+Requires:       python%{python3_pkgversion}-yaml
```
Comment (tchaikov) [inline ceph.spec.in]: could you conditionalize like
```
%if 0%{?fedora} || 0%{?rhel}
....
%if 0%{?rhel} == 7
Requires:       python%{python3_version_nodots}-PyYAML
%else
Requires:       python%{python3_version_nodots}-pyyaml
%endif
```
?

File: ceph.spec.in [hunk 2]
```diff
@@ -527,12 +529,14 @@ Requires:       python%{python3_pkgversion}-cherrypy
 Requires:       python%{python3_pkgversion}-jwt
 Requires:       python%{python3_pkgversion}-routes
 Requires:       python%{python3_pkgversion}-werkzeug
+Requires:       python%{python3_pkgversion}-yaml
```
Comment (tchaikov) [inline ceph.spec.in]: strange enough, i cannot find the place where dashboard imports `yaml`.

the only two places are
```
src/pybind/mgr/k8sevents/module.py:import yaml
src/pybind/mgr/orchestrator_cli/module.py:import yaml
```

File: ceph.spec.in (hunk 1 above)
Comment (liewegas) [reply ceph.spec.in]: it's not ``orchestrator``, and always-on.. so it needs to go in this package.

File: ceph.spec.in [hunk 3]
```diff
@@ -493,10 +493,16 @@ Requires:       python%{python3_pkgversion}-six
 %if 0%{?fedora} || 0%{?rhel}
 Requires:       python%{python3_pkgversion}-cherrypy
 Requires:       python%{python3_pkgversion}-werkzeug
+%if 0%{?rhel} == 7
+Requires:       python%{python3_pkgversion}-PyYAML
```
Comment (tchaikov) [inline ceph.spec.in]: s/python3_pkgversion /python3_version_nodots/

File: ceph.spec.in (hunk 1 above)
Comment (tchaikov) [reply ceph.spec.in]: `orchestrator` is an arch independent python module, that's why i think we should package in a separated package.

Comment (sebastian-philipp) [reply ceph.spec.in]: That's true for all mgr modules. If we want to have an arch independent package, we should package all mgr modules (except for rook,dashboard) in this new package.

**Comment by sebastian-philipp** [comment]
jenkins test make check

File: ceph.spec.in (hunk 1 above)
Comment (tchaikov) [reply ceph.spec.in]: see #33422

**Pull request merged**